cmd/raidz_test/raidz_test.c

```diff
@@ -0,0 +1,800 @@
+static kcondvar_t sem_cv;
+static int max_free_slots;
+static int free_slots;
+
+static void *
+sweep_thread(void *arg)
+{
+	int err = 0;
+	raidz_test_opts_t *opts = (raidz_test_opts_t *) arg;
+	VERIFY(opts != NULL);
+
+	err = run_test(opts);
+
+	if (rto_opts.rto_sanity) {
+		/* 25% chance that a sweep test fails */
+		if (rand() < (RAND_MAX/4))
+			err = 1;
+	}
+
+	if (0 != err) {
+		mutex_enter(&sem_mtx);
+		memcpy(&failed_opts, opts, sizeof (raidz_test_opts_t));
+		sweep_state = SWEEP_ERROR;
+		mutex_exit(&sem_mtx);
+	}
+
+	umem_free(opts, sizeof (raidz_test_opts_t));
+
+	/* signal the next thread */
+	mutex_enter(&sem_mtx);
+	free_slots++;
+	cv_signal(&sem_cv);
+	mutex_exit(&sem_mtx);
+
+	thread_exit();
+
+	return (NULL);
+}
+
+#define	ARR_LEN(x)	(sizeof (x) / sizeof (0[x]))
+
+static int
+run_sweep(void)
+{
+	static const size_t dcols_v[] = { 1, 2, 3, 4, 5, 6, 7, 8 };
+	static const size_t ashift_v[] = { 9, 12 };
+	static const size_t offset_cnt = 4;
+	static const size_t size_v[] = { 1<<9, 21 * (1<<9), 13 * (1<<12), 1<<17,
+	    (1<<20) - (1<<12), SPA_MAXBLOCKSIZE };
+
+	size_t total_comb = ARR_LEN(size_v) * ARR_LEN(ashift_v) *
+	    ARR_LEN(dcols_v) * offset_cnt;
+	size_t tried_comb = 0;
+	hrtime_t time_diff, start_time = gethrtime();
+	raidz_test_opts_t *opts;
+	int a, d, o, s;
+
+	max_free_slots = free_slots = MAX(2, sysconf(_SC_NPROCESSORS_ONLN));
+
+	mutex_init(&sem_mtx, NULL, MUTEX_DEFAULT, NULL);
+	cv_init(&sem_cv, NULL, CV_DEFAULT, NULL);
+
+	for (s = 0; s < ARR_LEN(size_v); s++)
+	for (a = 0; a < ARR_LEN(ashift_v); a++)
+	for (o = 0; o < offset_cnt; o++)
+	for (d = 0; d < ARR_LEN(dcols_v); d++) {
+
+		if ((size_v[s] < (1<<ashift_v[a]) * o) ||
+		    (size_v[s] < (1<<ashift_v[a]) * dcols_v[d])) {
+			total_comb--;
+			continue;
+		}
+
+		if (++tried_comb % 20 == 0)
+			PRINT("%lu/%lu... ", (ulong_t)tried_comb,
+			    (ulong_t)total_comb);
+
+		/* wait for signal to start new thread */
+		mutex_enter(&sem_mtx);
+		while (cv_timedwait_sig(&sem_cv, &sem_mtx,
+		    ddi_get_lbolt() + hz)) {
+
+			/* check if should stop the test (timeout) */
+			time_diff = (gethrtime() - start_time) / 1000000000ULL;
+			if (rto_opts.rto_sweep_timeout > 0 &&
+			    time_diff >= rto_opts.rto_sweep_timeout) {
+				sweep_state = SWEEP_TIMEOUT;
+				mutex_exit(&sem_mtx);
+				goto exit;
+			}
+
+			/* check if should stop the test (error) */
+			if (sweep_state != SWEEP_RUNNING) {
+				mutex_exit(&sem_mtx);
+				goto exit;
+			}
+
+			/* exit loop if a slot is available */
+			if (free_slots > 0) {
+				break;
+			}
+		}
+
+		free_slots--;
+		mutex_exit(&sem_mtx);
+
+		opts = umem_zalloc(sizeof (raidz_test_opts_t), UMEM_NOFAIL);
+		opts->rto_ashift = ashift_v[a];
+		opts->rto_dcols = dcols_v[d];
+		opts->rto_offset = (1<<ashift_v[a]) * o;
+		opts->rto_dsize = size_v[s];
+		opts->rto_v = 0; /* be quiet */
+
+		VERIFY3P(thread_create(NULL, 0,
+		    (thread_func_t) sweep_thread, (void *) opts,
+		    0, TS_RUN, 0, PTHREAD_CREATE_JOINABLE),
```

Comment:
These thread arguments are in the wrong order and the priority needs to be specified.  Something like this.

```
VERIFY3P(thread_create(NULL, 0, sweep_thread, opts, 0, curproc, TS_RUN, defclsyspri))
```

You'll also want to update the `sweep_thread()` prototype to return `void` so it matches the `thread_func_t`.  There's no need to return NULL from the thread function.